public/pages/workflow_detail/agentic_search/configure_flow/mcp_tool_filters.tsx

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React, { useState, useEffect } from 'react';
+import { getIn } from 'formik';
+import { EuiComboBox } from '@elastic/eui';
+import { MCPConnector } from '../../../../../common';
+
+interface MCPToolFiltersProps {
+  mcpServer: MCPConnector;
+  index: number;
+  updateMCPServer: (mcpServer: MCPConnector, index: number) => void;
+}
+
+export function MCPToolFilters(props: MCPToolFiltersProps) {
+  const [selectedFilters, setSelectedFilters] = useState<string[]>([]);
+  useEffect(() => {
+    setSelectedFilters(getIn(props, 'mcpServer.tool_filters', []) as string[]);
```

Comment:
Can we without getIn
```suggestion
    setSelectedFilters(props.mcpServer?.tool_filters ?? []) as string[]);
```

Reply:
Updated